nncf/common/logging/track_progress.py

```diff
@@ -0,0 +1,107 @@
+# Copyright (c) 2023 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Callable, Iterable, List, Optional, Sequence, Union
+
+from rich.console import Console
+from rich.progress import BarColumn
+from rich.progress import Progress
+from rich.progress import ProgressColumn
+from rich.progress import ProgressType
+from rich.progress import Task
+from rich.progress import TaskProgressColumn
+from rich.progress import TextColumn
+from rich.progress import TimeRemainingColumn
+from rich.style import StyleType
+from rich.text import Text
+
+
+class IterationsColumn(ProgressColumn):
+    def render(self, task: Task) -> Text:
+        if task.total is None:
+            return Text("")
+        text = f"{int(task.completed)}/{int(task.total)}"
+        if task.finished:
+            return Text(text, style="progress.elapsed")
+        return Text(text, style="progress.remaining")
+
+
+def track(
```

Comment:
And need add option to use progressbar in async mode to manually update iterations
https://github.com/openvinotoolkit/nncf/blob/19d7260310d07b6308e3bec6b585ed1a57054a05/tests/post_training/pipelines/image_classification_timm.py#L140-L156

Reply:
Added this functionality by turning `track` into a class with `__iter__`, `__enter__` and `__exit__` implemented similarly to `tqdm`. It can be used in the following way:
```
from nncf.common.logging.track_progress import progress
n = 10000
with track(total=n, description="Description") as pbar:
    for _ in range(n):
        np.random.random((100000,))
        pbar.progress.update(pbar.task, advance=1)
```